pkg/util/kubernetes/apiserver/leaderelection/leaderelection.go

```diff
@@ -243,13 +247,20 @@ func (le *LeaderEngine) GetLeader() string {
 
 // GetLeaderIP returns the IP the leader can be reached at, assuming its
 // identity is its pod name. Returns empty if we are the leader.
-// The result is not cached.
+// The result is cached and will not return an error if the leader does not exist anymore.
 func (le *LeaderEngine) GetLeaderIP() (string, error) {
```

Comment:
This is a breaking change with the current interface however I doubt that it will have a strong impact.
It used to always do an API Server Request to make sure that the leader is still up and running and retrieve its ip. Now it doesn't.
It means that:
- If the leader is down, it doesn't return an error when it should
- If the leader changed its IP (unlikely), it will return the old IP address.